---
title: "scRNA murine data analysis"
output: html_document
date: "2022-12-17"
---
### Requirements: 
R > 4.0.0
libraries from requirements.R script.
cloned repository with the pipeline https://github.com/califano-lab/single-cell-pipeline

```{r}
setwd('/home/daria/Rprojects/single-cell-pipeline')
.libPaths('./project_packages/')
```

```{r}
source('functions/process-utils.R')
source('functions/cluster-functions.R')
source('functions/viper-utils.R')
library(ggplot2)
library(ggpubr)
library(viper)
library(pheatmap)
library(RColorBrewer)
library(MUDAN)
library(umap)

raw.mat <- read.table('/home/daria/Rprojects/LV_SC_project/RPS007_matrix.txt',header = T,row.names = 1)
write.csv(rownames(raw.mat),"raw_names.csv",quote = F, row.names = F)
```
## pt 0: data review

First, the data is analyzed for quality control. Sequencing depth, number of detected genes, and mitochondrial gene percentage are assessed here. The PISCES pipeline provides a table of mitchondrial genes for both human and mouse samples;

murine mitochondrial genes were retrieved from here: http://asia.ensembl.org/biomart/martview/b2b79163486788b39f300cc958d90092

```{r, fig.align = 'center', fig.width = 10, fig.height = 5}
mt.genes <- read.table('/home/daria/Rprojects/LV_SC_project/murine_mito_genes.csv', header = TRUE, sep = ',', stringsAsFactors = FALSE)
mur.mt <- mt.genes$mur.symb
QCPlots(raw.mat, mur.mt)
```

This data is used to preprocess the data in several ways:

* Genes with no expression (zero reads across all cells) are removed
* Cells with less than 1000 UMIs or more than 100000 UMIs can be considered low quality, and are also removed.
* Additionally, if a population of cells has significantly more mitochondrial genes (generally, more than 10%), these cells can be removed with the *MTFilter* function.


The parameters for both the *QCTransform* and *MTFilter* function should be adjusted for your data.

```{r}
mt.mat <- MTFilter(raw.mat,mur.mt)
filt.mat <- QCTransform(mt.mat)
QCPlots(filt.mat, mur.mt)
```

Once the data set is filtered, it is normalized using CPM. Finally, a gene expression signature (*rank.mat*) is then generated:
By default, the gene expression signature is generated by performing a "double rank" transformation, which uses the median gene expression of the data set as an internal reference to compute a gene expression signature on a cell by cell basis.

```{r}
# run these if data was not normalized or transformed
#cpm.mat <- CPMTransform(filt.mat)
#rank.mat <- RankTransform(cpm.mat)

fin.mat <- filt.mat 
```


## pt 1: Integration analysis of human and mouse scRNA-seq data


The default PISCES approach assumes that there are no known cell type annotation in the data set. PISCES will generate an ARACNe network from all the cells. If different cell types in the data set are experimetally defined (e.g. by FACs) and annotated, cell-type specific networks can be generated based on the cell type annotation. However, we recommend proceeding in an unsupervised manner, as the unsupervised analysis can confirm the experimental design and, potentially, identify novel biological findings.

The data must first be saved in a format that is compatible with the Java based ARACNe-AP implementation included in this pipeline:

```{r}
library(dplyr)
ARACNeTable(filt.mat, '/home/daria/Rprojects/LV_SC_project/pbmc-cpm.tsv')
```

Reference-based integration analysis was performed on the ZsGreen1+ scRNA-seq data of Ldlr-
/- mouse fed 16-week WD and human atherosclerotic carotid artery scRNA-seq data, as well as
a public human coronary artery scRNA-seq dataset using Seurat v3.1.1. 
public dataset: https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE131778

Human gene symbols were first mapped to mouse gene symbols using biomaRt package in R and Ensembl 93
annotation. 
Integration anchors were identified using the same approach as described above,
except that the mouse dataset was specified as the reference.
```{r}

# load both count matrixes, human genes --> mouse genes via biomart, process, and then merge
# TBD; blocker -- biomart api is down, using custom functions

```

Cluster labels from mouse scRNA-seq joint analysis were used to identify human cells that clustered with mouse cells.
To investigate the counterpart of “fibromyocyte” population identified by Wirka et al.12 in
our mouse data, we first did clustering analysis of the human coronary scRNA-seq to identify
fibromyocyte population, then annotated each cell from the human coronary scRNA-seq dataset
using SingleR package in R with our scRNA-seq data of Ldlr-/- mice fed WD for 16 weeks as
reference.
```
Wirka et al 2019:
To localize these fibromyocytes within the plaque, we first searched for genes that were specific for the fibromyocyte cluster in the scRNAseq dataset. We found that the gene Lumican (Lum), when co-expressed with the tdT lineage marker, captured the majority of fibromyocytes with excellent (96%) specificity (Fig. 2k).
```
19.12 note: no TdT marker in the data at all, any synonyms? Or it should not be there?
```{r}
library(SingleR)
library(celldex)

ref.data <- MouseRNAseqData(cell.ont="all")# during first run it will load annotation to the cache
predictions <- SingleR(test=raw.mat, assay.type.test=1, 
    ref=ref.data, labels=ref.data$label.main)
table(predictions$pruned.labels)
```
## pt 2: Master regulators analysis
To infer master regulators (MRs) involved in cell state transition during atherosclerosis
development, we applied metaVIPER28, 29 on the ZsGreen1+ scRNA-seq data of Ldlr-/- mice fed
16-week WD, following the pipeline designed for MRs inference in scRNA-seq
(https://github.com/califano-lab/single-cell-pipeline).

Briefly, we first built a regulatory network using ARACNe on all single cells. ARACNe was run separately on each set of regulators: 1458 transcription factors, 821 transcriptional cofactors, and 3193 signaling proteins in mouse.

```{r}
ARACNeTable(fin.mat, '/home/daria/Rprojects/LV_SC_project/pbmc-cpm.tsv')
```

```
- install java and ANT
- build ARACNe-AP in separate folder (see instructions here: https://github.com/califano-lab/ARACNe-AP/blob/master/README.md)
- use script 'run_aracne.sh' in the terminal. Make sure that paths to the executables are correct. 

```


```{r}
netw_file = "/home/daria/Rprojects/LV_SC_project/ARACNA_out/merged_networks.tsv"
RegProcess(netw_file, rank.mat, out.dir = '/home/daria/Rprojects/LV_SC_project/ARACNA_out/', out.name = 'pbmc_r1-net-')
```

Gene expression signature was generated on each cell using the median gene expression as internal
reference. 

```{r}
r1.net <- readRDS('/home/daria/Rprojects/LV_SC_project/ARACNA_out/pbmc_r1-net-pruned.rds')
r1.pAct <- viper(rank.mat, r1.net, method = 'mad')
```

We then inferred protein activity based on the combined regulatory network and gene
expression signature. 


Partitioning around medoids (PAM) clustering was applied to distance matrix built on protein activity with the number of clusters ranging from 2 to 10. The optimal number of clusters was determined by silhouette score.
```{r}
r1.viperDist <- as.dist(viperSimilarity(r1.pAct))
r1.clusts <- PamKRange(r1.viperDist, kmin = 2, kmax = 10)
r1.clustSil <- SilScoreEval(r1.clusts, r1.viperDist)
plot.dat <- data.frame('k' = 2:10, 'Silhouette.Scores' = r1.clustSil)
ggplot(plot.dat, aes(x = k, y = Silhouette.Scores)) + geom_point() + geom_line() +
  ggtitle('1.1 Clustering Silhouette Scores') + theme_bw()
```
In this study, we have identified 3 clusters in mouse data based on protein activity. Next, we generated cluster-specific ARACNe networks using meta-cell profiles built from integrating gene expression profiles of cells with similar protein
activity profiles within each cluster. 
```{r}
r1.clustMats <- MakeCMfA(rank.mat, r1.viperDist, clustering = r1.clusts$k7, out.dir = '/home/daria/Rprojects/LV_SC_project/', out.name = 'pbmc-r1-clusts')
```
Counts from 5 nearest cells were integrated to generate each meta-cell. Protein activity was inferred again using cluster-specific networks as input to VIPER.
```{r}
c1.net <- RegProcess(netw_file , r1.clustMats[[1]], 
                     out.dir = '/home/daria/Rprojects/LV_SC_project/', out.name = 'pbmc-r2-c1_')
c2.net <- RegProcess(netw_file , r1.clustMats[[2]], 
                     out.dir = '/home/daria/Rprojects/LV_SC_project/', out.name = 'pbmc-r2-c2_')
c3.net <- RegProcess(netw_file , r1.clustMats[[3]], 
                     out.dir = '/home/daria/Rprojects/LV_SC_project/', out.name = 'pbmc-r2-c3_')
# load in networks
c1.net <- readRDS('/home/daria/Rprojects/LV_SC_project//pbmc-r2-c1_pruned.rds')
c2.net <- readRDS('/home/daria/Rprojects/LV_SC_project//pbmc-r2-c2_pruned.rds')
c3.net <- readRDS('/home/daria/Rprojects/LV_SC_project//pbmc-r2-c3_pruned.rds')
# infer protein activity
r2.pAct <- viper(rank.mat, list('c1' = c1.net, 'c2' = c2.net, 'c3' = c3.net), method = 'none')
```
T-test was conducted on transcription factor and cofactor activities between SMC and SEM
populations with 100 bootstraps. 

```{r}
r2.MRs <- BTTestMRs(r2.pAct, r2.louvain,bootstrap.num = 100)
```

Top activated and repressed MRs and their target genes are visualized using Cytoscape v3.3.0.
```{r}
BiocManager::install("RCy3")
library(Rcy3)
Top.MRS = MR_UnWrap(r2.MRs, top = 10)
#--> web
```

Protein activity in human unstable (visible zone of plaque rupture, n=4) and stable
(macroscopically normal adjacent areas, n=4, paired) atherosclerotic carotid artery plaques30
were estimated using metaVIPER. Prebuilt ARACNe networks from 45 tissues in GTEx
(https://github.com/califano-lab/GTEx-Networks) were used to infer protein activity in each
sample. metaVIPER is designed to accurately assess protein activities in tissues that do not have
a tissue-matched ARACNe network. Thus, networks built from hundreds of samples per tissue in
GTEx data are expected to provide accurate estimation of protein activities in atherosclerotic
plaque samples.
```{r}
# TBD
```
Gene expression from early (intimal thickening and xanthoma, n=13) and advanced
(fibrous cap atheroma, n=16) human atherosclerotic lesions of carotid arteries was downloaded
from Gene Expression Omnibus (accession number GSE28829)31. Protein activities in each
sample were estimated using metaVIPER with the same ARACNe networks from 45 tissues in
GTEx as described above.
```{r}
# TBD
```
